lib/Templating/Helper/BreachAttackRandomContent.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Enterprise License (PEL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+
+namespace Pimcore\Templating\Helper;
+
+use Symfony\Component\Templating\Helper\Helper;
+
+class BreachAttackRandomContent extends Helper
+{
+    /**
+     * @inheritDoc
+     */
+    public function getName()
+    {
+        return 'breachAttackRandomContent';
+    }
+
+    /**
+     * @return string
+     */
+    public function __invoke()
+    {
+        $length = 50;
+        $randomData = random_bytes($length);
+
+        return '<!--'
+            . substr(
+                base64_encode($randomData),
+                0,
+                ord($randomData[$length-1]) % 32
+            )
+            . '-->';
```

Comment:
This is a mitigation by increasing the number of attempts required, not the final fix, right?

Is there a long term plan to implement more complete protection?

Reply:
This is just an additional protection, the proper fix is to renew the CSRF token on every request.

Reply:
Done here: https://github.com/pimcore/pimcore/pull/5414/files#diff-85fefd7ecb60f6f6c28b38687d90c2aeR77